### DISCUSSION TITLE
How to turn off the background graphics in itwin-viewer

### DISCUSSION THREAD

**Discussion by JohnMajerle** [iTwin/itwinjs-core#1824] [Q&A] [answered]
I have created an iTwin Viewer using the template @bentley/itwin-viewer.  For debugging purposes I would like to not display the background satelite images because they obscure the items I am trying to view.

**Comment by bbastings** [comment, marked as answer]
A Viewport has a ViewFlag that controls whether to display the map. You can toggle it off/on as follows:  

const viewFlags = viewport.viewFlags.clone();
viewFlags.backgroundMap = onOff; // turn on or off...
viewport.viewFlags = viewFlags;

...you can also change where you set the default ViewFlags when setting up the view for your application to have it off by default.

**Comment by pmconne** [comment]
As an alternative to turning off the map, you can prevent it from obscuring your geometry by making it always draw in the background.
`viewport.changeBackgroundMapProps({ useDepthBuffer: false });`

**Comment by Carl-Hinkle** [comment]
@JohnMajerle @bentley/itwin-viewer has been deprecated and replaced with @itwin/web-viewer-react.  You can also bootstrap a new app with `npx create-react-app your-app-name --template @itwin/web-viewer --scripts-version @bentley/react-scripts`

**Reply by JohnMajerle** [reply]
Hey Carl,

Thanks for the heads up.  Is **_@itwin/web-viewer-react_** significantly different from **_@bentley/itwin-viewer_**?  If so in what way?

Thanks,

John M.

**Reply by Carl-Hinkle** [reply]
It was ~~just~~ mainly an organizational change.  The viewer needed to be moved the the @itwin org and the name was changed to be consistent with the new @itwin/desktop-viewer-react package.  The minimum dependency versions were also updated.

Thanks,
Carl

**Reply by calebmshafer** [reply]
The one major change is the removal of desktop support within the single `@bentley/itwin-viewer` package. That has now been split out into it's own package as @Carl-Hinkle references above. However the web functionality is consistent across both the new and old package.
